web/src/main/java/org/springframework/security/web/access/intercept/RequestMatcherDelegatingAuthorizationManager.java

```diff
@@ -127,11 +134,49 @@ public Builder add(RequestMatcher matcher, AuthorizationManager<RequestAuthoriza
 		 */
 		public Builder mappings(
 				Consumer<List<RequestMatcherEntry<AuthorizationManager<RequestAuthorizationContext>>>> mappingsConsumer) {
+			Assert.state(!this.anyRequestConfigured, "Can't configure mappings after anyRequest");
 			Assert.notNull(mappingsConsumer, "mappingsConsumer cannot be null");
 			mappingsConsumer.accept(this.mappings);
 			return this;
 		}
 
+		/**
+		 * Maps an array of string patterns to {@link AuthorizationManager}.
+		 * @param patterns the patterns to map
+		 * @return the {@link AuthorizedUrl} for further customizations
+		 * @since 6.1
+		 */
+		public AuthorizedUrl requestMatchers(String... patterns) {
```

Comment:
The `requestMatchers` method should keep the same behavior as `AbstractRequestMatcherRegistry` by creating a `MvcRequestMatcher` if possible and use `AntPathRequestMatcher` as a fallback. https://github.com/spring-projects/spring-security/blob/1631cac1506d7fdda1f3864357075af4965a78cd/config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java#L181

Reply:
Hi @marcusdacoregio, thanks for the review. I'm not sure if it would be a good idea to propagate an `ApplicationContext` to this builder, maybe we could create here `AntPathRequestMatcher` by default
and let the user register `MvcRequestMatcher` using `requestMatchers(RequestMatcher... matchers)`.

What do you think?

Reply:
I think you are right, can you make that behavior explicit in the javadoc?

Reply:
I've reflected this behavior in the javadoc and updated `@since` tags to `6.2`.

Reply:
Hi @evgeniycheban, I am concerned about having the ant matcher the default if Spring MVC is in the classpath. This deviates from the default that Spring Security adopted in 6.0. For now, what if `requestMatchers` only takes a `RequestMatcher` instance? We can think more (in another issue) about how to communicate the MVC nuance to the builder.

Reply:
Hi @marcusdacoregio, makes sense, let's stay only with `requestMatchers(RequestMatcher... matchers)` for now and think about supporting `MvcRequestMatcher` in the future releases. I will update the PR soon.

Reply:
I've updated the PR.